### PR TITLE
feat(DIST-779): React lib

### PR DESCRIPTION
**Pull request by mathio**

### CODE REVIEW

**Review by antoniofull — APPROVED** [review]

File: packages/embed-react/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      //   use: [
+      //     {
+      //       loader: require('styled-jsx/webpack').loader,
+      //       options: {
+      //         type: 'scoped',
+      //       },
+      //     },
+      //   ],
+      //   sideEffects: true,
+      // },
```
Comment (antoniofull) [inline packages/embed-react/webpack.config.js:30]: Is this needed?

Comment (mathio) [reply packages/embed-react/webpack.config.js:30]: Good catch, I think it is not.

**Review by scarciofolomarco — APPROVED** [review]

File: packages/embed-react/src/components/make-button-component.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const ref: MutableRefObject<GenericEmbed | null> = useRef(null)
+    useEffect(() => {
+      ref.current = createFn(id, props)
+      return () => ref.current?.unmount()
+    }, [id, props])
+
+    return (
+      <>
+        <InlineStyle filename={cssFilename} />
+        <button onClick={() => ref.current?.open()} className={className} style={style}>
```
Comment (mrisoli) [inline packages/embed-react/src/components/make-button-component.tsx:34]: Will the arrow function here not cause a rerender on every prop change?

File: packages/embed-react/src/components/make-button-component.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+type CreateFn<T> = (id: string, props: CreateFnProps<T>) => GenericEmbed
+
+type GenericEmbed = {
+  unmount: () => void
+  open: () => void
+}
+
+function makeButtonComponent<T>(createFn: CreateFn<T>, cssFilename: string) {
+  return ({ id, children, style = {}, className = '', ...props }: ButtonComponentProps<T>) => {
+    const ref: MutableRefObject<GenericEmbed | null> = useRef(null)
```
Comment (mrisoli) [inline packages/embed-react/src/components/make-button-component.tsx:25]: I'm not a fan of the union with null here, I'd consider maybe having a default object instead, not saying necessarily that it is better, probably a matter of taste

File: packages/embed/src/factories/create-popup/create-popup.ts [hunk 4]
```diff
@@ -97,7 +98,7 @@ export const createPopup = (formId: string, userOptions: PopupOptions = {}): Pop
       popup.style.opacity = '0'
       wrapper.style.opacity = '0'
       setTimeout(() => {
-        popup.parentNode.removeChild(popup)
+        popup.parentNode && popup.parentNode.removeChild(popup)
```
Comment (mrisoli) [inline packages/embed/src/factories/create-popup/create-popup.ts:101]: Could be replaced with a safe navigation operator here:

```suggestion
        popup.parentNode?.removeChild(popup)
```

Comment (scarciofolomarco) [reply packages/embed/src/factories/create-popup/create-popup.ts:101]: theres the `unmountElement` function too

File: packages/embed-react/src/components/make-button-component.tsx (hunk 3 above)
Comment (mathio) [reply packages/embed-react/src/components/make-button-component.tsx:25]: Something like this?

```
const emptyEmbed: GenericEmbed = {
  unmount: () => {},
  open: () => {},
}
...
const ref: MutableRefObject<GenericEmbed> = useRef(emptyEmbed)
```

**Comment by rix1** [comment]
Looking very much forward to having this merged! 🤩 Let me know if you need beta users 😁

**Review by prilutskiy — APPROVED** [review]

**Comment by mathio** [comment]
React lib is released, please follow [the README](https://github.com/Typeform/embed/tree/main/packages/embed-react) for installation and usage. If there is anything to report, [please open a Github issue](https://github.com/Typeform/embed/issues). I will appreciate any testing.

cc: @rix1

**Pull request merged**